bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/handler/WemoHandler.java

```diff
@@ -311,68 +340,59 @@ public void onValueReceived(@Nullable String variable, @Nullable String value, @
         }
     }
 
-    private synchronized void onSubscription() {
-        if (service.isRegistered(this)) {
-            logger.debug("Checking WeMo GENA subscription for '{}'", this);
-
-            ThingTypeUID thingTypeUID = thing.getThingTypeUID();
-            String subscription = "basicevent1";
+    private synchronized void addSubscription() {
+        synchronized (upnpLock) {
+            if (service.isRegistered(this)) {
+                logger.debug("Checking WeMo GENA subscription for '{}'", this);
 
-            if (subscriptionState.get(subscription) == null) {
-                logger.debug("Setting up GENA subscription {}: Subscribing to service {}...", getUDN(), subscription);
-                service.addSubscription(this, subscription, SUBSCRIPTION_DURATION_SECONDS);
-                subscriptionState.put(subscription, true);
-            }
+                ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+                String subscription = "basicevent1";
 
-            if (thingTypeUID.equals(THING_TYPE_INSIGHT)) {
-                subscription = "insight1";
                 if (subscriptionState.get(subscription) == null) {
                     logger.debug("Setting up GENA subscription {}: Subscribing to service {}...", getUDN(),
                             subscription);
                     service.addSubscription(this, subscription, SUBSCRIPTION_DURATION_SECONDS);
                     subscriptionState.put(subscription, true);
                 }
+
+                if (thingTypeUID.equals(THING_TYPE_INSIGHT)) {
+                    subscription = "insight1";
```

Comment:
Extract to constant for consistency as this is used in multiple places.

Reply:
Whay constant ???

Reply:
Will change it later on...

Reply:
Done